documentation/performance/container-tuning.html

```diff
@@ -17,157 +18,60 @@ <h2 id="container-worker-threads">Container worker threads</h2>
-defaults are most often good for production - not a development
-machine. This is not an extensive guide on how to tune the JVM,
-but rather a list of pointers of where to look and change.
-</p><p>
-If things work when running Vespa, skip reading here. However,
-if parts of Vespa fail at startup or while running, and it
-might be memory related, look below for hints.
-Example <em>vespa.log</em> entries:
-<pre>
-  1222175132.631524       mynode.mydomain.com   12213   config-sentinel config-sentinel.service event   stopped/1 name="qrserver" pid=15414 exitcode=32768
-  1222175132.631583       mynode.mydomain.com   12213   config-sentinel config-sentinel.service error   qrserver: Attempted to start, but fork() failed: Cannot allocate memory
-  
-  ...
-  
-  1222175132.52427        mynode.mydomain.com   15381   logserver       stdout  info    Java HotSpot(TM) 64-Bit Server VM warning: Attempt to deallocate stack guard pages failed.
-  1222175132.52480        mynode.mydomain.com   15381   logserver       stdout  info    Java HotSpot(TM) 64-Bit Server VM warning: Attempt to allocate stack guard pages failed.
-  1222175132.036          mynode.mydomain.com   -/10    -       ADM.com.yahoo.logserver.handlers.HandlerThread  config  logserver.queue.size=200
-  1222175132.52521        mynode.mydomain.com   15381   logserver       stderr  warning Exception in thread "main" java.lang.OutOfMem
-  oryError: unable to create new native thread
-  
-  ...
-  
-  1222175130.810528       mynode.mydomain.com   14314   logserver       stderr  warning dl failure on line 685 Error: failed $VESPA_HOME/libexec64/jdk1.6.0/jre/lib/amd64/server/libjvm.so, because $VESPA_HOME/libexec64/jdk1.6.0/jre/lib/amd64/server/libjvm.so: failed to map segment from shared object: Cannot allocate memory
-</pre>
-A frequent case is trying to run some a sample Vespa application,
-configured with all services, on a development machine -
-this consumes gigabytes of memory and might fail.
-</p><p>
-Set JVM parameters for Java services in <em>services.xml</em>.
-When modified, <em>vespa-deploy prepare</em>, <em>vespa-deploy activate</em> and restart the relevant services.
-Example: Run two containers with concurrent garbage collection:
+Use <em>jvm-gc-options</em> for controlling GC related parameters and <em>jvm-options</em> for tuning other parameters. See <a href="../reference/services-container.html#nodes">reference documentation</a>. 
+Example: Running with CMS concurrent garbage collection and using NewRatio = 1 (equal size of old and new generation) and enabling verbose GC logging (Logged to stdout to vespa.log file).
 <pre>
 &lt;container id="default" version="1.0"&gt;
-  &lt;nodes jvmargs="-XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+CMSIncrementalPacing"&gt;
+  &lt;nodes jvm-gc-options="-XX:+UseConcMarkSweepGC -XX:MaxTenuringThreshold=15 -XX:NewRatio=1 -XX:+PrintGC" jvm-options="-XX:+PrintCommandLineFlags"&gt;
```

Comment:
The reference doc is wrong then? https://docs.vespa.ai/documentation/reference/services-container.html#nodes - says jvmargs only - I guess jvm-gc-options is added recently?

Reply:
it's also in this PR to add those two new options added some time back but not documented.

Reply:
disregard - I see you fixed this. CREDIT - good job!